TodoApi/Extensions/SerilogExtensions.cs

```diff
@@ -0,0 +1,68 @@
+using Microsoft.Extensions.Options;
+using Serilog;
+using Serilog.Exceptions;
+
+namespace TodoApi;
+
+public static class SerilogExtensions
+{
+    public static WebApplicationBuilder AddSerilog(
+        this WebApplicationBuilder builder,
+        string sectionName = "Serilog")
+    {
+        builder.Services.AddOptions<SerilogOptions>()
+            .BindConfiguration(sectionName)
+            .ValidateDataAnnotations();
+
+        builder.Host.UseSerilog((context, serviceProvider, loggerConfiguration) =>
+        {
+            var loggerOptions = serviceProvider.GetRequiredService<IOptions<SerilogOptions>>().Value;
+
+            // https://github.com/serilog/serilog-settings-configuration
+            loggerConfiguration.ReadFrom.Configuration(context.Configuration, sectionName: sectionName);
+
+            loggerConfiguration
+                .Enrich.WithProperty("Application", builder.Environment.ApplicationName)
+                .Enrich.FromLogContext()
+                .Enrich.WithEnvironmentName()
+                .Enrich.WithMachineName()
+                // https://rehansaeed.com/logging-with-serilog-exceptions/
+                .Enrich.WithExceptionDetails();
+
+            if (loggerOptions.UseConsole)
+            {
+                // https://github.com/serilog/serilog-sinks-async
+                loggerConfiguration.WriteTo.Async(writeTo =>
+                    writeTo.Console(outputTemplate: loggerOptions.LogTemplate));
+            }
```

Comment:
On interesting thing about this is the fact that we would need to remove the other logging providers probably. I don't know if UseSerilog removes the other providers by default though. Maybe it does. Is there a reason to turn this off?